task-sdk/src/airflow/sdk/definitions/taskgroup.py

```diff
@@ -247,14 +246,14 @@ def add(self, task: DAGNode) -> DAGNode:
         if isinstance(task, TaskGroup):
             if self.dag:
                 if task.dag is not None and self.dag is not task.dag:
-                    raise RuntimeError(
+                    raise ValueError(
                         "Cannot mix TaskGroups from different Dags: %s and %s",
                         self.dag,
                         task.dag,
                     )
                 task.dag = self.dag
             if task.children:
-                raise AirflowException("Cannot add a non-empty TaskGroup")
+                raise ValueError("Cannot add a non-empty TaskGroup")
```

Comment:
Same with this one? I can imagine someone doing this:
```python
    with DAG("test_dag", schedule=None, start_date=pendulum.parse("20200101")):
        with TaskGroup("section_with_tasks") as tg_with_children:
            task1 = EmptyOperator(task_id="task1")

        parent_tg = TaskGroup("parent_section")
        parent_tg.add(tg_with_children)
```

This would have thrown AirflowException earlier, now its ValueError

Reply:
I think this one is fine, as it would have been a dag import error before, so I can't see anything beyond the tiniest of edge cases that someone was catching this before.